### ISSUE TITLE
UnicodeDecodeError when reading SnapGene files

### ISSUE THREAD

**Issue by nicolazilio** [biopython/biopython#3180]
### Setup

I am reporting a problem with Biopython version, Python version, and operating system as follows:

```
>>> import sys; print(sys.version)
3.6.8 (default, May 24 2019, 19:22:50) 
[GCC 6.3.0 20170516]
>>> import platform; print(platform.python_implementation()); print(platform.platform())
CPython
Linux-4.19.0-9-amd64-x86_64-with-debian-10.4
>>> import Bio; print(Bio.__version__)
1.77
```

### Expected behaviour

I would like to get feature names for > 4000 SnapGene plasmid maps. Using SnapGene Server, I can retrieve these data from all plasmid maps. I would expect the same using BioPython.

### Actual behaviour

BioPython fails to retrieve feature names for > 80% of the plasmid maps due to errors similar to the following

```
File "/home/lib/python3.6/site-packages/Bio/SeqIO/SnapGeneIO.py", line 119, in _parse_features_packet
    xml = parseString(data.decode("ASCII"))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 709: ordinal not in range(128)
```

### Possible solution

I amended `Bio/SeqIO/SnapGeneIO.py` to replace all occurrences of `ASCII` to `utf8`. This fixes the exception shown above. Is there a good reason for using `ASCII` rather than `utf8`, given that biologists very often use special characters and the SnapGene format does support them?

**Comment by peterjc** [comment]
Please review #2831, and if you could provide a test case we can include with Biopython that would be really helpful.

**Comment by nicolazilio** [comment]
Ah, thanks! I searched for issues containing `SnapGene` but didn't see this one. I'm including a plasmid with a double whammy :), it contains a ® and a β. Both of the features that contain these characters are part of SnapGene's collection of common features. 
[pFA-KanMX4.zip](https://github.com/biopython/biopython/files/5000084/pFA-KanMX4.zip)

**Comment by peterjc** [comment]
This should be fixed on the master branch thanks to #3187 - independent confirmation testing would be most welcome (please comment here and reopen it if needed).
